apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -1,71 +1,60 @@
-import os
 from abc import ABC, abstractmethod, abstractproperty
 from typing import Optional, Union, cast
 
 import tiledb
 
 from . import util
 from .metadata_mapping import MetadataMapping
-from .tiledb_platform_config import TileDBPlatformConfig
+from .soma_tiledb_context import SOMATileDBContext
 
 
 class TileDBObject(ABC):
     """
     Base class for ``TileDBArray`` and ``Collection``.
 
-    Manages tiledb_platform_config, context, etc. which are common to both.
+    Accepts a SOMATileDBContext, to enable session state to be shared across SOMA objects.
     """
 
     _uri: str
-    _tiledb_platform_config: TileDBPlatformConfig
+    _context: SOMATileDBContext
     _metadata: MetadataMapping
 
     def __init__(
         self,
         # All objects:
         uri: str,
         *,
-        # Non-top-level objects can have a parent to propgate context, depth, etc.
+        # Non-top-level objects can have a parent to propagate context, depth, etc.
         parent: Optional["TileDBObject"] = None,
-        # Top-level objects should specify these:
-        tiledb_platform_config: Optional[TileDBPlatformConfig] = None,
-        ctx: Optional[tiledb.Ctx] = None,
+        # Top-level objects should specify this:
+        context: Optional[SOMATileDBContext] = None,
     ):
         """
-        Initialization-handling shared between ``TileDBArray`` and ``Collection``.  Specify ``tiledb_platform_config`` and ``ctx`` for the top-level object; omit them and specify parent for non-top-level objects. Note that the parent reference is solely for propagating options, ctx, display depth, etc.
+        Initialization-handling shared between ``TileDBArray`` and ``Collection``.  Specify ``context`` for
+        the top-level object; omit it and specify parent for non-top-level objects. Note that the parent reference
+        is solely for propagating the context
         """
 
-        if ctx is None:
-            ctx = self._default_ctx()
         self._uri = uri
-        if parent is None:
-            self._ctx = ctx
-        else:
-            tiledb_platform_config = parent._tiledb_platform_config
-            self._ctx = parent._ctx
 
-        self._tiledb_platform_config = tiledb_platform_config or TileDBPlatformConfig()
-        # Null ctx is OK if that's what they wanted (e.g. not doing any TileDB-Cloud ops).
+        if parent is not None:
+            assert (
+                context is None
+            ), "Only one of `context` and `parent` params can be passed as an arg"
```

Comment:
`assert` statements should only ever be used for internal sanity checks, so:

```suggestion
            if context is not None:
                raise TypeError("Only one of `context` and `parent` params can be passed as an arg")
```